program/rust/src/accounts.rs

```diff
@@ -109,7 +113,6 @@ pub trait PythAccount: Pod {
         )?;
 
         check_valid_fresh_account(account)?;
-        clear_account(account)?;
```

Comment:
maybe worth a code comment that the solana runtime guarantees that the account content is all zero here.

(feel free to do in a later pr because i know the CI takes forever here)